keras/applications/resnet_rs.py

```diff
@@ -0,0 +1,957 @@
+
+        x = layers.Add()([x, shortcut])
+
+        return layers.Activation(activation, name=name + "_output_act")(x)
+
+    return apply
+
+
+def BlockGroup(
+    filters,
+    strides,
+    num_repeats,
+    se_ratio: float = 0.25,
+    bn_epsilon: float = 1e-5,
+    bn_momentum: float = 0.0,
+    activation: str = "relu",
+    survival_probability: float = 0.8,
+    name=None,
+):
+    """Create one group of blocks for the ResNet model."""
+    if name is None:
+        counter = backend.get_uid("block_group_")
+        name = f"block_group_{counter}"
+
+    def apply(inputs):
+        # Only the first block per block_group uses projection shortcut and strides.
+        x = BottleneckBlock(
+            filters=filters,
+            strides=strides,
+            use_projection=True,
+            se_ratio=se_ratio,
+            bn_epsilon=bn_epsilon,
+            bn_momentum=bn_momentum,
+            activation=activation,
+            survival_probability=survival_probability,
+            name=name + "_block_0_",
+        )(inputs)
+
+        for i in range(1, num_repeats):
+            x = BottleneckBlock(
+                filters=filters,
+                strides=1,
+                use_projection=False,
+                se_ratio=se_ratio,
+                activation=activation,
+                bn_epsilon=bn_epsilon,
+                bn_momentum=bn_momentum,
+                survival_probability=survival_probability,
+                name=name + f"_block_{i}_",
+            )(x)
+        return x
+
+    return apply
+
+
+def get_survival_probability(init_rate, block_num, total_blocks):
+    """Get survival probability based on block number and initial rate."""
+    return init_rate * float(block_num) / total_blocks
+
+
+def allow_bigger_recursion(target_limit: int):
+    """Increase default recursion limit to create larger models."""
+    current_limit = sys.getrecursionlimit()
+    if current_limit < target_limit:
+        sys.setrecursionlimit(target_limit)
+
+
+def fixed_padding(inputs, kernel_size):
+    """Pad the input along the spatial dimensions independently of input size."""
+    pad_total = kernel_size - 1
+    pad_beg = pad_total // 2
+    pad_end = pad_total - pad_beg
+
+    # Use ZeroPadding as to avoid TFOpLambda layer
+    padded_inputs = layers.ZeroPadding2D(
+        padding=((pad_beg, pad_end), (pad_beg, pad_end))
+    )(inputs)
+
+    return padded_inputs
+
+
+def ResNetRS(
+    depth: int,
+    input_shape=None,
+    bn_momentum=0.0,
+    bn_epsilon=1e-5,
+    activation: str = "relu",
+    se_ratio=0.25,
+    dropout_rate=0.25,
+    drop_connect_rate=0.2,
+    include_top=True,
+    block_args: List[Dict[str, int]] = None,
+    model_name="resnet-rs",
+    pooling=None,
+    weights="imagenet",
+    input_tensor=None,
+    classes=1000,
+    classifier_activation: Union[str, Callable] = "softmax",
+    include_preprocessing=True,
+):
+    """Build Resnet-RS model, given provided parameters.
+
+    Args:
```

Comment:
unindent docstring to match code indentation

Reply:
Changed indentation from 4 spaces to 2 spaces, it should be good now. Let me know if anything, need to be done.